retrofit/src/test/java/retrofit/CallTest.java

```diff
@@ -424,18 +424,16 @@
         .build();
     Service example = retrofit.create(Service.class);
 
-    server.enqueue(new MockResponse().setBody("1234").throttleBody(1, 500, MILLISECONDS));
+    server.enqueue(new MockResponse()
+        .setBody("1234")
+        .setSocketPolicy(DISCONNECT_DURING_RESPONSE_BODY));
 
-    long exeuteStart = System.nanoTime();
-    Response<ResponseBody> response = example.getBody().execute();
-    long executeTook = System.nanoTime() - exeuteStart;
-    assertThat(executeTook).isGreaterThan(MILLISECONDS.toNanos(1000));
-
-    long readStart = System.nanoTime();
-    String body = response.body().string();
-    long readTook = System.nanoTime() - readStart;
-    assertThat(readTook).isLessThan(MILLISECONDS.toNanos(500));
-    assertThat(body).isEqualTo("1234");
+    // When buffering we will detect all socket problems before returning the Response.
+    try {
+      example.getBody().execute();
```

Comment:
Maybe extract `getBody()` out of the try/catch to make it clear that this only happens on `execute()` ?